lib/connection.cpp

```diff
@@ -901,17 +902,11 @@ Room* Connection::roomByAlias(const QString& roomAlias, JoinStates states) const
 }
 
 void Connection::updateRoomAliases(const QString& roomId,
-                                   const QString& aliasServer,
                                    const QStringList& previousRoomAliases,
                                    const QStringList& roomAliases)
 {
-    auto& aliasMap = d->roomAliasMap[aliasServer]; // Allocate if necessary
-    for (const auto& a : previousRoomAliases)
-        if (aliasMap.remove(a) == 0)
-            qCWarning(MAIN) << "Alias" << a << "is not found (already deleted?)";
-
-    for (const auto& a : roomAliases) {
-        auto& mappedId = aliasMap[a];
+    for (const auto& a: roomAliases) {
+        auto& mappedId = d->roomAliasMap[a];
```

Comment:
You seem to have dropped an important part here that removes old aliases from the list. The way you have your code now, if some alias from `d->roomAliasMap` is in `previousRoomAliases` but not in `roomAliases` it seems to get stuck in `d->roomAliasMap` forever.